mice-and-cats-java/src/main/java/foop/Client.java

```diff
@@ -76,54 +79,35 @@ public void run() {
                     jFrame.dispose();
                 }
             });
-            System.out.println(playerName + " exited");
+            log.info("{} exited", playerName);
         }
     }
 
     private void createAndShowGUI() {
-        JFrame f = new JFrame(playerName);
-        jFrame = f;
-        f.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
-        f.addWindowListener(new WindowAdapter() {
-            @Override
-            public void windowClosed(WindowEvent e) {
-                Client.this.running = false;
-                try {
-                    Client.this.socket.close();
-                } catch (IOException ex) {
-                    ex.printStackTrace();
-                }
-            }
-        });
-        f.setSize(1000, 500);
-
-        f.setLayout(new GridLayout());
-        f.add(new JComponent() {
-            @Override
-            public void paint(Graphics graphics) {
-                repaint(); // already request next repaint
-
-                Graphics2D g = (Graphics2D) graphics;
-                var w = Client.this.world;
-                if (w != null) {
-                    w.render(g, getWidth(), getHeight());
-                } else {
-                    g.clearRect(0, 0, getWidth(), getHeight());
-                }
-            }
-        });
 
-        var lobbyListModel = new DefaultListModel<AvailableGamesMessage.Game>();
-        var lobbyList = new JList<>(lobbyListModel);
-        lobbyList.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
-        f.add(lobbyList);
+        this.jFrame = new GameFrame(this);
+        this.updateLobby = updateLobby();
+    }
 
-        updateLobby = (message) -> {
+    @Override
+    public void close() throws Exception {
+        this.socket.close();
+    }
+
+
+    @SneakyThrows
+    public synchronized void send(Message message) {
+        serialize(message, this.socket.getOutputStream());
+    }
+
+    public synchronized Consumer<AvailableGamesMessage> updateLobby() {
+        return availableGamesMessage -> {
```

Comment:
Why return a lambda and store it in a member variable instead of just defining `private synchronized void updateLobby(AvailableGamesMessage m)`?